### PR TITLE
Feature: gacha command

### PR DESCRIPTION
**Pull request by danerprog**
Added !gacha command that allows the bot to roll for a command given a user input.

The commands to roll and the user input is configured via gacha.json and can be enabled or disabled via options.ini

### CODE REVIEW

File: config/example_gacha.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+	"binatog" : {
+		"play as if its your last blackpink" : 50,
+		"play binatog" : 50
+	}
+}
```
Comment (Seijuro1010) [inline config/example_gacha.json]: very minor: add new line at the end of the file

File: musicbot/bot.py [hunk 2]
```diff
@@ -3880,19 +3886,22 @@ async def on_message(self, message):
...
-            else:
+            modified_command = self.command_modifier.modifyUsingAlias(command)
+            modified_command, *modified_args = modified_command.split(" ")
+            
+            if modified_command == "gacha":
+                modified_command = self.command_modifier.modifyUsingGacha(" ".join(modified_args))
+                modified_command, *modified_args = modified_command.split(" ")
+            elif modified_command == "" and command == "gacha":
+                modified_command = self.command_modifier.modifyUsingGacha(" ".join(args))
+                modified_command, *modified_args = modified_command.split(" ")
```
Comment (Seijuro1010) [inline musicbot/bot.py]: Suggestion: Process these inside ModifiedCommand, so we'll only need to replace `alias_command = self.aliases.get(command)` in the original code

File: config/example_gacha.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+	"binatog" : {
+		"play as if its your last blackpink" : 50,
+		"play binatog" : 50
+	}
```
Comment (Seijuro1010) [inline config/example_gacha.json:5]: Suggestion: Add proper parameter labels/names at it is being interpreted as json.

File: musicbot/gacha.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        if gacha_command in self._gacha_commands:
+            normal_command = self._rollNormalCommand(gacha_command)
+        log.debug("normal_command: " + str(normal_command))
+        return normal_command
+                
+
+class GachaDefault:
+    gacha_file = "config/gacha.json"
+    gacha_seed = {}
+    gacha = {}
```
Comment (Seijuro1010) [inline musicbot/gacha.py:133]: Extra line may be needed

File: musicbot/CommandModifier.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            self._gacha = NoGacha()
+        else:
+            self._gacha = Gacha(gacha_file)
+            
+    def modifyUsingAlias(self, command):
+        return self._aliases.get(command)
+        
+    def modifyUsingGacha(self, command):
+        return self._gacha.roll(command)
+
```
Comment (Seijuro1010) [inline musicbot/CommandModifier.py]: Extra line may be needed

**Review by Seijuro1010 — APPROVED** [review]

File: config/example_gacha.json (hunk 3 above)
Comment (danerprog) [reply config/example_gacha.json:5]: no need imo. this is the same format as aliases.json where the key is the command

**Review by Seijuro1010 — APPROVED** [review]

**Pull request merged**